app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", only: :destroy 
+  
+  def create
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.create(comments_params)
+
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.find(params[:id])
```

Comment:
How can you find the `Comment` without finding the `Article` in the first place? What is the best approach and why?

Reply:
I can delete the comment directly using the class method find:

`@comment = Comment.find(params[:id])`

The best approach is finding the article and deleting the comment because this ensures that we deleted the comment that belongs to the article and finally returns the updated article.

Reply:
If i use
`@comments = Comment.find(params[:id])` 
i only make a query to delete the comment, with `params[:id_article]` is verified if this comment belong to the article.

Reply:
```
def destroy
    if @comment.article_id == params[:article_id].to_i
      @comment.destroy
    end
    redirect_back fallback_location: articles_url
 end
```